File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/Tab.java
package org.dolphinemu.dolphinemu.ui.main;

import org.dolphinemu.dolphinemu.R;

public enum Tab
{
  GAMECUBE(0, "GameCube"),
  WII(1, "Wii"),
  WIIWARE(2, "Wii Ware"),
  MORE(3, "More");

  private final int value;
  private final String idString;

  Tab(int value, String idString)
  {
    this.value = value;
    this.idString = idString;
  }

  public int toInt()
  {
    return value;
  }

  public String getIdString()
  {
    return idString;
  }

  public static int getTabId(int tab)
  {
    if (tab == Tab.GAMECUBE.toInt())
    {
      return R.id.action_gamecube;
    }
    else if (tab == Tab.WII.toInt())
    {
      return R.id.action_wii;
    }
    else if (tab == Tab.WIIWARE.toInt())
    {
      return R.id.action_wiiware;
    }
    else if (tab == Tab.MORE.toInt())
    {
      return R.id.action_more;
    }
    return 0;
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/Tab.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.ui

import org.dolphinemu.dolphinemu.R

enum class Tab(private val value: Int) {
    GAMECUBE(0),
    WII(1),
    WIIWARE(2),
    MORE(3);

    fun toInt(): Int {
        return value
    }

    fun getID(): Int {
        when (value) {
            GAMECUBE.toInt() -> return R.id.gamecubeFragment
            WII.toInt() -> return R.id.wiiFragment
            WIIWARE.toInt() -> return R.id.wiiwareFragment
            MORE.toInt() -> return R.id.moreFragment
        }
        return 0
    }

    companion object {
        fun fromInt(i: Int): Tab {
            return Tab.values()[i]
        }
    }
}
